plugins/modules/swim_workflow_manager.py

```diff
@@ -1679,6 +1681,15 @@ def get_diff_tagging(self):
                     self.result['response'] = self.msg
                     self.log(self.msg, "INFO")
                     break
+                elif task_details.get("isError"):
```

Comment:
getting isError, progress, failure_reason from task_details used in multiple places.  
Can't  we get the vaules, store and use it?

<<< Sample Code, modify the code based on your logic.... >>>>>>>>>>

```
device_family = tagging_details.get("device_image_family_name")
device_role = tagging_details.get("device_role", "ALL")
site_name = tagging_details.get("site_name", "Global")

while True:
    task_details = self.get_task_details(task_id)
    is_error = task_details.get("isError")
    progress = task_details.get("progress", "")
    failure_reason = task_details.get("failureReason", "")

    if is_error:
        if not tag_image_golden and "An inheritted tag cannot be un-tagged" in failure_reason:
            self.msg = failure_reason
        else:
            action = "Tagging" if tag_image_golden else "Un-Tagging"
            self.msg = (
                "{0} image {1} golden for site {2} for family {3} for device role {4} failed."
                .format(action, image_name, site_name, device_family, device_role)
            )
        self.status = "failed"
        self.result['changed'] = False
        self.result['msg'] = self.msg
        self.log(self.msg, "ERROR")
        break

    if "successful" in progress:
        action = "Tagging" if tag_image_golden else "Un-Tagging"
        self.msg = (
            "{0} image {1} golden for site {2} for family {3} for device role {4} successful."
            .format(action, image_name, site_name, device_family, device_role)
        )
        self.status = "success"
        self.result['changed'] = True
        self.result['msg'] = self.msg
        self.log(self.msg, "INFO")
        break

return self

```

What happens if not is_error, and not successful in progress? Do we sleep for some time? Is it needed to be in while loop to do tag and untag an image? If we still need to check "while" loop, then add sleep and do checks for default times and break from the loop..